azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -146,19 +137,7 @@ public boolean apply(final String name) {
 
       final VMDeployment deployment = deployments.iterator().next();
 
-
-      NodeAndInitialCredentials<VMDeployment> credential = null;
-
-      if (template.getOptions().getPublicKey() != null){
-         String privateKey = template.getOptions().getPrivateKey();//this can be null as it will use the key privated in ssh-agent
-         credential = new NodeAndInitialCredentials<VMDeployment>(deployment, name,
-         LoginCredentials.builder().user(loginUser).privateKey(privateKey).authenticateSudo(true).build());
-      } else {
-         credential = new NodeAndInitialCredentials<VMDeployment>(deployment, name,
-         LoginCredentials.builder().user(loginUser).password(loginPassword).authenticateSudo(true).build());
-      }
-
-      return credential;
+      return new NodeAndInitialCredentials<VMDeployment>(deployment, name, null);
```

Comment:
It is safe to pass `null` credentials here, as jclouds will default populate the node with the default credentials from the image, or the ones in the options, if the user provided them. This is done [here](https://github.com/jclouds/jclouds/blob/master/compute/src/main/java/org/jclouds/compute/strategy/impl/AdaptingComputeServiceStrategies.java#L198).